apps/system/js/notifications.js

```diff
@@ -284,12 +289,20 @@ var NotificationScreen = {
       return;
     }
 
-    if (Math.abs(this._touchPosX) >
+    // Taking speed into account to detect swipes
+    var deltaT = Date.now() - this._touchStartDate;
+    var speed = this._touchPosX / deltaT;
+    var inertia = speed * this.SWIPE_INERTIA;
+    var finalDelta = Math.abs(this._touchPosX + inertia);
+
+    if (finalDelta >
         this._containerWidth * this.TRANSITION_FRACTION) {
       if (this._touchPosX < 0) {
         this._notification.classList.add('left');
       }
-      this.swipeCloseNotification();
+
+      var durationLeft = (1 - (finalDelta / this._containerWidth)) * 300;
+      this.swipeCloseNotification(durationLeft);
```

Comment:
Nice :) Might be nice to factor out the 300 into a const so this reads better (and we can document it a little better)